ocaml/xcp-rrdd/scripts/rrdd/rrdd.py

```diff
@@ -223,7 +223,8 @@ def lazy_complete_init(self):
         self.path = self.dispatcher.get_path({"uid": self.uid})
         base_path = os.path.dirname(self.path)
         if not os.path.exists(base_path): os.makedirs(base_path)
-        self.dest = open(self.path, "wb")
+        fd = os.open(self.path, os.O_RDWR|os.O_CREAT)
+        self.dest = os.fdopen(fd, mode='r+b')
```

Comment:
I suggest adding a variant of Pau's comment in the PR above:

> add:
> "Open file at pos 0, creating but not truncating the file(not possible otherwise):"

```suggestion
        # Open file at pos 0, creating but not truncating the file(not possible otherwise):
        fd = os.open(self.path, os.O_RDWR|os.O_CREAT)
        self.dest = os.fdopen(fd, mode='r+b')
```
PS: Reviewers can ignore the GitHub annotation `attribute-defined-outside-init` for this PR:
Such `pylint` clean-up can be done later, taking care of any code relying on AttributeError that may rely on it to not be an attribute until it is set (Yes, we've such ugly code still around...)